File: user/src/main/java/com/kjj/user/service/manager/ManagerService.java
package com.kjj.user.service.manager;

import com.kjj.user.dto.info.ManagerInfoDto;
import com.kjj.user.entity.User;
import com.kjj.user.exception.CantFindByUsernameException;
import com.kjj.user.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class ManagerService {

    private final UserRepository userRepository;

    @Transactional(readOnly = true)
    public ManagerInfoDto getInfo(String username) throws CantFindByUsernameException {
        User user = userRepository.findByUsername(username).orElseThrow(() -> new CantFindByUsernameException("""
                username을 가진 유저 데이터가 없습니다.
                username : """ + username));
        return ManagerInfoDto.from(user);
    }
}

File: user/src/main/kotlin/com/kjj/user/service/manager/ManagerService.kt
package com.kjj.user.service.manager

import com.kjj.user.dto.info.ManagerInfoDto
import com.kjj.user.exception.CantFindByIdException
import com.kjj.user.repository.user.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ManagerService(
    val userRepository: UserRepository
) {
    @Transactional(readOnly = true)
    fun getInfo(username: String): ManagerInfoDto {
        val user = userRepository.findByUsername(username)
            ?: throw CantFindByIdException(
                    """
                    username을 가진 유저 데이터가 없습니다.
                    username : $username
                    """
            )

        return ManagerInfoDto.from(user)
    }
}